### DISCUSSION TITLE
LS 2.3.0+git@968eb6ff5 + Autocue

### DISCUSSION THREAD

**Discussion by gAlleb** [Moonbase59/autocue#34] [General]
I've tried [968eb6ff5fe02f4528754070ae2488f96429de05](https://github.com/savonet/liquidsoap/commit/968eb6ff5fe02f4528754070ae2488f96429de05) and now running it with `autocue 4.1.0`. 

So far so good.

Had to make minor changes to `autocue.cue_file.liq` and comment out everything connected to `settings.autocue.target_cross_duration` and it works. See https://github.com/gAlleb/autocue/commit/cb6bdfdb3fde78d6f13dbedadce0d2784b7226da for example.

Waiting for @Moonbase59 to look at new LS.

 
<img width="1219" alt="image" src="https://github.com/user-attachments/assets/76c90425-33da-494a-b1eb-b80190d5fbf8">

**Comment by Moonbase59** [comment]
Maybe I can look into this mid/end August, or when LS 2.3.0 is finished. (See availability note in README.)

Metadata names have changed again (see https://github.com/savonet/liquidsoap/blob/968eb6ff5fe02f4528754070ae2488f96429de05/src/libs/autocue.liq#L892-L911), and the `cross_duration` changes to `cross_start_duration`, `cross_end_duration` and `cross_max_start_duration`. `cross_start_next` seems to have been replaced by `fade_out_start_next` (?).

These changes are upstream and out of our control, so
- a new Autocue version will definitely be needed for LS 2.3.0
- removing old/adding new tags in pre-tagged files will be necessary.

If you feel like experimenting in the meantime, check out the new [`cross` operator](https://github.com/savonet/liquidsoap/blob/968eb6ff5fe02f4528754070ae2488f96429de05/src/core/operators/cross.ml#L443-L519).

**Reply by gAlleb** [reply]
True. Have a nice vacation!
